### ISSUE TITLE
[plugin-typescript] - Svelte import not resolved when "node" types is set in tsconfig

### ISSUE THREAD

**Issue by mgrisole** [rollup/plugins#864]
- Rollup Plugin Name: @rollup/plugin-typescript
- Rollup Plugin Version: 8.0.0
- Rollup Version: 2.3.4
- Operating System (or Browser): macOs BigSur
- Node Version: 14.16.0
- Link to reproduction: https://replit.com/@mgrisole/svelte-typescript-playground#tsconfig.json

### Expected Behavior
Rollup should be able to resolve .svelte imports.

### Actual Behavior
When adding "node" to the types's compiler option it prevents typescript from resolving .svelte imports. 
Getting the following error on build:

`(!) Plugin typescript: @rollup/plugin-typescript TS2307: Cannot find module './App.svelte' or its corresponding type declarations.`

### Additional Information
To be honest I am not sure if this issue issue is directly related to this plugin but it is my best bet at the moment.

![image](https://user-images.githubusercontent.com/33571141/115484869-8aba0780-a221-11eb-9114-9d5d43a1a4d2.png)

**Comment by mgrisole** [comment]
After some digging I found the solution. 

----
### TL;DR
Don't forget to include svelte in your types definition:  `"types": ["node", "svelte"]`

----

I omitted the extended _tsconfig.json_ provided by Svelte which include "svelte" as a type, therefore I was overwriting the type by only setting "node" in my base _tsconfig.json._

Weirdly enough I had another project with the following types definition and it was working fine: `"types": ["node", "jest"]`
